mqtt_client/src/MqttClient.ros2.cpp

```diff
@@ -88,7 +194,10 @@ bool primitiveRosMessageToString(
   } else if (msg_type == "std_msgs/msg/Bool") {
     std_msgs::msg::Bool msg;
     deserializeRosMessage(*serialized_msg, msg);
-    primitive = msg.data ? "true" : "false";
+    if (fixed_encoding)
+      primitive = msg.data ? "1" : "0";
+    else
+      primitive = msg.data ? "true" : "false";

```

Comment:
Would you mind again explaining this reply of yours? Is it related to this part of the code?

> I should also be forward that there is a ninja nerf in here that is kinda ugly.
> Whenever we are dealing with fixed_types I changed the encodings of bools in the primitive mqtt messages to be "0" or "1" instead of "true" or "false".
> In my particular use case this is because this is how a device I am communicating with encodes those things and I couldn't figure out a clean way to make either or supported without a bunch of typing.